control-plane/connect-inject/container_init_test.go

```diff
@@ -340,156 +248,43 @@ func TestHandlerContainerInit_transparentProxy(t *testing.T) {
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-		},
-		"enabled globally, ns not set, annotation is false": {
-			globalEnabled: true,
-			annotations:   map[string]string{keyConsulDNS: "false"},
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-		},
-		"enabled globally, ns not set, annotation is true": {
-			globalEnabled: true,
-			annotations:   map[string]string{keyConsulDNS: "true"},
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -consul-dns-ip="10.0.34.16" \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-		},
-		"disabled globally, ns not set, annotation not provided": {
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-		},
-		"disabled globally, ns not set, annotation is false": {
-			annotations: map[string]string{keyConsulDNS: "false"},
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-		},
-		"disabled globally, ns not set, annotation is true": {
-			annotations: map[string]string{keyConsulDNS: "true"},
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -consul-dns-ip="10.0.34.16" \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-		},
-		"disabled globally, ns enabled, annotation not set": {
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -consul-dns-ip="10.0.34.16" \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-			namespaceLabel: map[string]string{keyConsulDNS: "true"},
-		},
-		"enabled globally, ns disabled, annotation not set": {
-			globalEnabled: true,
-			expectedContainsCmd: `/consul/connect-inject/consul connect redirect-traffic \
-  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
-  -proxy-uid=5995`,
-			namespaceLabel: map[string]string{keyConsulDNS: "false"},
-		},
-	}
-	for name, c := range cases {
-		t.Run(name, func(t *testing.T) {
-			w := MeshWebhook{
-				EnableConsulDNS:        c.globalEnabled,
-				EnableTransparentProxy: true,
-				ResourcePrefix:         "consul-consul",
-				ConsulConfig:           &consul.Config{HTTPPort: 8500},
 			}
-			os.Setenv("CONSUL_CONSUL_DNS_SERVICE_HOST", "10.0.34.16")
-			defer os.Unsetenv("CONSUL_CONSUL_DNS_SERVICE_HOST")
-
-			pod := minimal()
-			pod.Annotations = c.annotations
-
-			ns := testNS
-			ns.Labels = c.namespaceLabel
-			container, err := w.containerInit(ns, *pod, multiPortInfo{})
-			require.NoError(t, err)
-			actualCmd := strings.Join(container.Command, " ")
-
-			require.Contains(t, actualCmd, c.expectedContainsCmd)
-		})
-	}
-}
-
-func TestHandler_constructDNSServiceHostName(t *testing.T) {
```

Comment:
this test has been moved to the `redirect_traffic_test.go` because that's where this logic now lives